apps/api/src/modules/providers/vc-twilio/vc-twilio.service.ts

```diff
@@ -0,0 +1,171 @@
+  machineDetectionSpeechEndThreshold?: number;
+  machineDetectionSilenceTimeout?: number;
+  asyncAmd?: string;
+  asyncAmdStatusCallback?: string;
+  asyncAmdStatusCallbackMethod?: string;
+  byoc?: string;
+  callReason?: string;
+  callToken?: string;
+  recordingTrack?: string;
+  timeLimit?: number;
+  url?: string;
+  twiml?: string;
+  applicationSid?: string;
+}
+
+export interface VcTwilioResponseData {
+  account_sid: string;
+  annotation: string | null;
+  answered_by: string | null;
+  api_version: string;
+  caller_name: string | null;
+  date_created: string;
+  date_updated: string;
+  direction: string;
+  duration: string;
+  end_time: string;
+  forwarded_from: string;
+  from: string;
+  from_formatted: string;
+  group_sid: string | null;
+  parent_call_sid: string | null;
+  phone_number_sid: string;
+  price: string;
+  price_unit: string;
+  sid: string;
+  start_time: string;
+  status: string;
+  subresource_uris: {
+    notifications: string;
+    recordings: string;
+    payments: string;
+    events: string;
+    siprec: string;
+    streams: string;
+    transcriptions: string;
+    user_defined_message_subscriptions: string;
+    user_defined_messages: string;
+  };
+  to: string;
+  to_formatted: string;
+  trunk_sid: string | null;
+  uri: string;
+  queue_time: string;
+}
+
+@Injectable()
+export class VcTwilioService {
+  private twilioClient;
+  private twilioVoiceCallObject: Partial<VcTwilioData>;
+
+  constructor(private readonly providersService: ProvidersService) {}
+
+  async assignTransport(providerId: number): Promise<void> {
+    const vcTwilioConfig = await this.providersService.getConfigById(providerId);
+    const accountSid = vcTwilioConfig.TWILIO_VC_ACCOUNT_SID as string;
+    const authToken = vcTwilioConfig.TWILIO_VC_AUTH_TOKEN as string;
+    this.twilioClient = Twilio(accountSid, authToken);
+  }
+
+  async filterRequestBody(requestBody: Partial<VcTwilioData>): Promise<void> {
+    const allowedKeys = [
+      'from',
+      'to',
+      'method',
+      'fallbackUrl',
+      'fallbackMethod',
+      'statusCallback',
+      'statusCallbackEvent',
+      'statusCallbackMethod',
+      'sendDigits',
+      'timeout',
+      'record',
+      'recordingChannels',
+      'recordingStatusCallback',
+      'recordingStatusCallbackMethod',
+      'sipAuthUsername',
+      'sipAuthPassword',
+      'machineDetection',
+      'machineDetectionTimeout',
+      'recordingStatusCallbackEvent',
+      'trim',
+      'callerId',
+      'machineDetectionSpeechThreshold',
+      'machineDetectionSpeechEndThreshold',
+      'machineDetectionSilenceTimeout',
+      'asyncAmd',
+      'asyncAmdStatusCallback',
+      'asyncAmdStatusCallbackMethod',
+      'byoc',
+      'callReason',
+      'callToken',
+      'recordingTrack',
+      'timeLimit',
+      'url',
+      'twiml',
+      'applicationSid',
+    ];
+
+    if (requestBody.url === undefined && requestBody.twiml === undefined) {
```

Comment:
Is this not redundant if we already have a check in DTO?

Reply:
Removed this check